frb/galaxies/eazy.py

```diff
@@ -0,0 +1,363 @@
+        newv.append(str(fluxtable[key].data[0]))
+    with open(catfile, 'w') as f:
+        # Filters
+        allf = ' '.join(newfs)
+        f.write('# {} \n'.format(allf))
+        # Values
+        f.write(' '.join(newv))
+    print("Wrote catalog file: {}".format(catfile))
+    base_cat = os.path.basename(catfile)
+
+    # Input file
+    default_file = os.path.join(resource_filename('frb', 'data'), 'analysis', 'EAZY', 'zphot.param.default')
+    with open(default_file, 'r') as df:
+        df_lines = df.readlines()
+
+    with open(param_file, 'w') as f:
+        for dfline in df_lines:
+            if 'CATALOG_FILE' in dfline:
+                line = dfline.replace('REPLACE.cat', base_cat)
+            elif prior_filter is not None and 'APPLY_PRIOR' in dfline:
+                line = dfline.replace('n', 'y', 1)
+            elif prior_filter is not None and 'PRIOR_FILTER' in dfline:
+                line = dfline.replace('999', str(frb_to_eazy_filters[prior_filter]), 1)
+            elif prior_filter is not None and 'PRIOR_FILE' in dfline:
+                line = dfline  # Deal with this if we do anything other than r
+            elif prior_filter is not None and 'PRIOR_ABZP' in dfline:
+                line = dfline  # Deal with this if we do anything other than r
+            elif 'Directory to put output files in' in dfline:  # Relative to the Input directory
+                line = dfline[0:10]+dfline[10:].replace('OUTPUT', out_dir, -1)
+            else:
+                line = dfline
+            # Write
+            f.write(line)
+    print("Wrote param file: {}".format(param_file))
+
+
+def run_eazy(input_dir, name, logfile):
+    """
+    Find and run the EAZY executable on the files
+
+    Args:
+        input_dir (str):
+            Path to eazy inputs/ folder  (can be relative)
+            This is where eazy will be run
+        name (str):
+            Name of the source being analzyed
+        logfile (str):
+    """
+    _, param_file = eazy_filenames(input_dir, name)
+
+    # Find the eazy executable
+    path_to_eazy = spawn.find_executable('eazy')
+    if path_to_eazy is None:
+        raise ValueError("You must have eazy in your Unix path..")
+    # Run it!
+    command_line = [path_to_eazy, '-p', os.path.basename(param_file)]
+    with open(logfile, 'w') as f:
+        retval = subprocess.call(command_line, stdout=f, stderr=f, cwd=input_dir)
+
+
+def eazy_stats(zgrid, pzi):
+    """
+    Calculate the 'best' zphot and error
+
+    Args:
+        zgrid (np.ndarray):
+        pzi (np.ndarray):
+
+    Returns:
+
+    """
+    # p(z) weighted
+    zphot = np.sum(zgrid * pzi) / np.sum(pzi)
+    # Uncertainty
+    cum_pzi = np.cumsum(pzi) / np.sum(pzi)
+    l68 = zgrid[np.argmin(np.abs(cum_pzi - 0.16))]
+    u68 = zgrid[np.argmin(np.abs(cum_pzi - (1 - 0.16)))]
+    #
+    return zphot, (u68-l68)/2.
+
+
+####################################################
+#  The following are taken from threedhst by Brummer
```

Comment:
Do consider adding the SED fit visualization functions from there as well.

Reply:
considered, but will pass for now.